### DISCUSSION TITLE
Maintaining Git for saleor origin and my private origin remote repository

### DISCUSSION THREAD

**Discussion by tiolierobin** [saleor/saleor#12216] [Q&A] [answered]
Hi Saleor Communities! currently I am planning to create self hosted ecommerce website for my own business. I want to add new payment gateway integration to the code or maybe adding another feature outside saleor core. Any idea how can I maintain git remote between saleor repository and my private repository? assuming I want to update saleor core from saleor repository to the latest version and also updating my own code in the future
thanks in advance !

**Comment by karolkielecki** [comment, marked as answer]
Hi @tiolierobin, 

Could you please share more details about your business use case?

Have you tried [Saleor Cloud](https://cloud.saleor.io/signup)? You can create a free developer account to test it. If you go with our Cloud, you don't need to worry about maintaining & updating your instance to the latest version. 

Generally speaking, forking the code is not recommended because that can lead to technical debt and issues with maintaining and updating to the latest versions. 

Regarding integrations such as payment gateways, plugins will be depreciated soon. The recommended way is to create an app. Apps are not part of the code. You can find more details in our [documentation](https://docs.saleor.io/docs/3.x/developer/extending/apps/key-concepts).
